client/lib/query-manager/media/test/index.js

```diff
@@ -0,0 +1,339 @@
+			it( 'should return false if query is not ISO 8601', () => {
+				const isMatch = manager.matches( {
+					before: '2018'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.false;
+			} );
+
+			it( 'should return false if media is not before date', () => {
+				const isMatch = manager.matches( {
+					before: '2010-04-25T15:47:33-04:00'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.false;
+			} );
+
+			it( 'should return true if media is before date', () => {
+				const isMatch = manager.matches( {
+					before: '2018-04-25T15:47:33-04:00'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.true;
+			} );
+		} );
+
+		context( 'query.after', () => {
+			it( 'should return false if query is not ISO 8601', () => {
+				const isMatch = manager.matches( {
+					after: '2010'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.false;
+			} );
+
+			it( 'should return false if media is not after date', () => {
+				const isMatch = manager.matches( {
+					after: '2018-04-25T15:47:33-04:00'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.false;
+			} );
+
+			it( 'should return true if media is after date', () => {
+				const isMatch = manager.matches( {
+					after: '2010-04-25T15:47:33-04:00'
+				}, DEFAULT_MEDIA );
+
+				expect( isMatch ).to.be.true;
+			} );
+		} );
+	} );
+
+	describe( '#sort()', () => {
+		context( 'query.order', () => {
+			it( 'should sort descending by default', () => {
+				const sorted = [
+					{ ID: 200 },
+					{ ID: 400 }
+				].sort( manager.sort.bind( manager, {
+					order_by: 'ID'
+				} ) );
+
+				expect( sorted ).to.eql( [
+					{ ID: 400 },
+					{ ID: 200 }
+				] );
+			} );
+
+			it( 'should reverse order when specified as ascending', () => {
+				const sorted = [
+					{ ID: 200 },
+					{ ID: 400 }
+				].sort( manager.sort.bind( manager, {
+					order_by: 'ID',
+					order: 'ASC'
+				} ) );
+
+				expect( sorted ).to.eql( [
+					{ ID: 200 },
+					{ ID: 400 }
+				] );
+			} );
+		} );
+
+		context( 'query.order_by', () => {
+			context( 'date', () => {
+				const olderMedia = {
+					...DEFAULT_MEDIA,
+					date: '2016-04-25T11:40:52-04:00'
+				};
+
+				const newerMedia = {
+					...DEFAULT_MEDIA,
+					ID: 152,
+					date: '2016-04-25T15:47:33-04:00'
+				};
+
+				it( 'should order by date', () => {
+					const sorted = [
+						olderMedia,
+						newerMedia
+					].sort( manager.sort.bind( manager, {} ) );
+
+					expect( sorted ).to.eql( [
+						newerMedia,
+						olderMedia
+					] );
+				} );
+			} );
+
+			context( 'title', () => {
+				const aMedia = {
+					...DEFAULT_MEDIA,
+					title: 'a'
+				};
+
+				const zMedia = {
+					...DEFAULT_MEDIA,
+					ID: 152,
+					title: 'z'
+				};
+
+				it( 'should sort by title', () => {
+					const sorted = [
+						aMedia,
+						zMedia
+					].sort( manager.sort.bind( manager, {
+						order_by: 'title'
+					} ) );
+
+					expect( sorted ).to.eql( [
+						zMedia,
+						aMedia
+					] );
+				} );
+			} );
```

Comment:
>A test with mixed case titles (e.g. "a" and "B") could be useful in order to verify that case is not taken into consideration when comparing titles.

This made me curious to see what the result would be. It appears `localeCompare` does not consider capitalization as a typical `>` comparison might:

![image](https://cloud.githubusercontent.com/assets/1779930/21015984/f9e90072-bd31-11e6-999b-5b9dc6589dbb.png)

I tried testing this in a live environment and it seems that WordPress will always lower-case the title of an uploaded image, even if the file itself has mixed casing.

I agree though that adding a test here would be wise.